inst/mlr3shiny/server/server_evaluating.R

```diff
@@ -0,0 +1,246 @@
+# reactive values for the chosen learner, the features selected and the iml objects used for explaining the model
+eval_meta <- reactiveValues(current_learner = NULL, selected_features = NULL, feature_effect = NULL, feature_importance = NULL, feature_importance_compare = NULL)
+
+# available losses for multiclass/twoclass classification and regression tasks
+classif_losses <- c("ce", "f1")
+regr_losses <- c(
+  "mae", "mse", "rmse", "mape",
+  "mdae", "msle", "percent_bias", "rae", "rmsle", "rse", "rrse", "smape"
+)
+twoclass_losses <- c("logLoss")
+
+# shiny outputs for the ui
+output$eval_learner_selection <- renderUI({
+  get_learner_selection()
+})
+
+output$eval_loss_function_selection <- renderUI({
+  get_loss_function_list()
+})
+
+output$eval_compare_method_selection <- renderUI({
+  get_compare_method_list()
+})
+
+output$eval_learner_feat_selection <- renderUI({
+  get_feature_selection()
+})
+
+output$eval_learner_plot_tabs <- renderUI({
+  display_plot_tabs()
+})
+
+# observe "start evaluation" button and start iml workflow
+observeEvent(input$evaluate_start, {
+  withCallingHandlers(
+    tryCatch({
+        model <- Predictor$new(eval_meta$current_learner, data = currenttask$task$data(), y = currenttask$task$target_names)
+        # saving iml calculations in meta object
+        eval_meta$feature_importance <- FeatureImp$new(model, loss = input$loss_picker, compare = input$compare_picker)
+        eval_meta$feature_effect <- FeatureEffects$new(model, method = "pdp")
+        calculate_plots()
+      },
+      error = errorAlertTrain
+    )
+  )
+})
+
+# observe choosen learner in ui and change meta object
+observeEvent(input$selected_learner, {
+  reset_evaluation()
+  eval_meta$current_learner <- trained_learner_list[[input$selected_learner]]
+  # showing only relevant loss functions
+  reset_plots()
+})
+
+# save selected features and reset plot diagrams
+observeEvent(input$feat_picker, {
+  eval_meta$selected_features <- input$feat_picker
+})
+
+# builder for compare method selection
+get_compare_method_list <- function() {
+  if (!is.null(input$selected_learner)) {
+    ui <- tagList(
+      fluidRow(
+        column(
+          12,
+          h5("Select compare method for Feature Importance:")
+        ),
+        column(
+          12,
+          pickerInput("compare_picker",
+            choices = c("ratio", "difference")
+          )
+        )
+      )
+    )
+    return(ui)
+  }
+}
+
+
+# get relevant loss functions for current task
+get_loss_function_list <- function() {
```

Comment:
Please declare function parameters that are used to avoid implicit variable changes